### PR TITLE
anna_lewis_1708_scrabble

### PR DESCRIPTION
**Pull request by anlewis**
needs refactoring :(
but it works :D

### CODE REVIEW

File: test/scrabble_test.rb [hunk 1]
```diff
@@ -5,9 +5,41 @@
...
+
   def test_it_can_score_a_single_letter
-    skip
     assert_equal 1, Scrabble.new.score("a")
     assert_equal 4, Scrabble.new.score("f")
   end
+
+  def test_it_can_score_a_word
+    assert_equal 4, scrabble.score("anna")
+    assert_equal 8, scrabble.score("word")
```
Comment (VictoriaVasys) [inline test/scrabble_test.rb:26]: I like that you tested multiple words; could have improved robustness (is that a word? :) ) by testing words of different length.

File: test/scrabble_test.rb [hunk 2]
```diff
@@ -5,9 +5,41 @@
...
+  end
+
+  def test_it_can_score_an_empty_string
+    assert_equal 0, scrabble.score("")
+  end
+
+  def test_it_can_score_nil
```
Comment (VictoriaVasys) [inline test/scrabble_test.rb:33]: I like that you split out edge case tests!

File: test/scrabble_test.rb [hunk 3]
```diff
@@ -5,9 +5,41 @@
...
+
+  def test_it_scores_with_letter_multipliers
+    assert_equal 9, scrabble.score_with_multipliers('hello', [1,2,1,1,1])
+    assert_equal 18, scrabble.score_with_multipliers('hello', [1,2,1,1,1], 2)
+  end
+
+  def test_it_gives_10_point_bonus
+    assert_equal 48, scrabble.score_with_multipliers('sparkle', [1,2,1,3,1,2,1], 2)
+  end
 end
```
Comment (VictoriaVasys) [inline test/scrabble_test.rb:45]: Would like to see more tests to reassure that your code is working properly; test different lengths, different multipliers on letters & the whole word, etc.

File: lib/scrabble.rb [hunk 4]
```diff
@@ -1,7 +1,48 @@
 class Scrabble
 
+  def split_word(word)
+    word.chars
+  end
+
+  def capitalize_letters(letters)
```
Comment (VictoriaVasys) [inline lib/scrabble.rb:7]: This method & the next are great use-cases for map & show a strong effort to break out logic!

File: lib/scrabble.rb [hunk 5]
```diff
@@ -1,7 +1,48 @@
...
+
+  def check_10_point_bonus(multiplied, letters)
+    if letters.count < 7
+      multiplied.sum
+    elsif letters.count >= 7
+      (multiplied.sum + 10)
+    end
+  end
+
+  def score_with_multipliers(word, multipliers, final_multiplier = 1)
```
Comment (VictoriaVasys) [inline lib/scrabble.rb:35]: I'd love for you to identify ways you could refactor this method; it's a bit long and difficult to read.

**Review by VictoriaVasys — COMMENTED** [review]
Overall, well done Anna! Way to make many commits throughout and break out logic. Continue to work to refactor so that each method maintains only its responsibility & is readable. I'd also love to see you add tests to ensure more robust implementation code (this also indicates to other developers that you were thinking about these things; e.g. your code is dynamic).
